src/engine/client/hunk_allocator.cpp

```diff
@@ -325,6 +325,7 @@ void Hunk_FreeTempMemory( void *buf )
 
 	if ( hdr->magic != (int) HUNK_MAGIC )
 	{
+		ASSERT( false );
```

Comment:
This is not a good way to do it as we use some compiler builtins to instruct the compiler to assume that the assert condition is always true. So the compiler may remove the Sys::Error.

Maybe you should put something like `if (IsDebuggerAttached) BREAKPOINT;` inside of Sys::Error to solve the problem of catching errors in the debugger.

Reply:
Done.

Actually we may discuss the topic of always adding a breakpoing within `Sys::Error` instead, but this is good enough for now.